lib/bolt/inventory/group2.rb

```diff
@@ -152,28 +155,35 @@ def add_target(target)
         unless target.is_a?(Hash)
           raise ValidationError.new("Node entry must be a Hash, not #{target.class}", @name)
         end
+
         # This check prevents plugins from returning plugins
         raise ValidationError.new("Cannot set target with plugin", @name) if target.key?('_plugin')
         target.each do |k, v|
           next if k == 'config'
           validate_config_plugin(v, k, @name)
         end
 
-        target['name'] ||= target['uri']
+        unless target['name']
+          target['name'] = target['uri']
```

Comment:
Is there a reason we can't just use a variable here instead of putting the `name` field on the target? Then we don't have to delete it below. I may be missing somewhere that we use the whole target hash though...

Reply:
We do in fact put it in `t_name`, but only at the end of the function. Can we do that at the top of the function and refer to `t_name` everywhere?

Reply:
Oh, yeah, that makes way more sense. That is a much better approach. Updated.